app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeightForm.java

```diff
@@ -0,0 +1,196 @@
+		inchInput = contentView.findViewById(R.id.maxHeightInchInput);
+	}
+
+	private int getMaxHeightLayoutResourceId()
+	{
+		return isMetric ? R.layout.quest_max_height : R.layout.quest_max_height_us;
+	}
+
+	private void addOtherAnswers()
+	{
+
+		if (getCountryInfo().getHeightUnits().size() == 2)
+		{
+			if (isMetric)
+			{
+				addOtherAnswer(R.string.quest_maxheight_answer_imperial_unit, () ->
+				{
+					isMetric = false;
+					setMaxHeightSignLayout(getMaxHeightLayoutResourceId());
+				});
+			}
+			else
+			{
+				addOtherAnswer(R.string.quest_maxheight_answer_metric_unit, () ->
+				{
+					isMetric = true;
+					setMaxHeightSignLayout(getMaxHeightLayoutResourceId());
+				});
+			}
+		}
+
+		addOtherAnswer(R.string.quest_maxheight_answer_noSign, () ->
+		{
+			new AlertDialogBuilder(getActivity())
+				.setTitle(R.string.quest_maxheight_answer_noSign_confirmation_title)
+				.setMessage(R.string.quest_maxheight_answer_noSign_confirmation)
+				.setPositiveButton(R.string.quest_maxheight_answer_noSign_confirmation_positive, (dialog, which) -> {
+					Bundle answer = new Bundle();
+					answer.putBoolean(NO_SIGN, true);
+					applyImmediateAnswer(answer);
+				})
+				.setNegativeButton(R.string.quest_generic_confirmation_no, null)
+				.show();
+		});
+	}
+
+	@Override protected void onClickOk()
+	{
+		if(!hasChanges())
+		{
+			Toast.makeText(getActivity(), R.string.no_changes, Toast.LENGTH_SHORT).show();
+			return;
+		}
+
+		if(userSelectedUnrealisticHeight())
+		{
+			confirmUnusualInput(this::applyMaxHeightFormAnswer);
+		}
+		else
+		{
+			applyMaxHeightFormAnswer();
+		}
+	}
+
+	private boolean userSelectedUnrealisticHeight()
+	{
+		double height = Double.parseDouble(getHeightFromInput());
+		double heightInMeter = isMetric ? height : feetToMeter(height);
+		return heightInMeter > 6;
+	}
+
+	private static double feetToMeter(double feet)
+	{
+		return feet / 3.2808;
+	}
+
+	private void applyMaxHeightFormAnswer()
+	{
+		Bundle answer = new Bundle();
+
+		String height = getFinalMaxHeight(getHeightFromInput());
+
+		// metric is the OSM default, does not need to be mentioned
+		if(!isMetric)
+		{
+			//this adds an apostrophe and a double-quote to be in a format like e.g. 6'7"
+			height = height.replace(".", "'");
+			height += "\"";
+		}
+		answer.putString(MAX_HEIGHT, height);
+
+		applyFormAnswer(answer);
+	}
+
+	private String getFinalMaxHeight(String value)
+	{
+		if (isMetric)
+		{
+			/*TODO: should a trailing zero be added or does it even work without?
+			https://taginfo.openstreetmap.org/keys/maxheight#values contains both...*/
+			if (!value.contains("."))
+			{
+				value += ".0";
+			}
+			return value;
+		}
+		else
+		{
+			return value.replace(".", "'") + "\"";
+		}
+	}
+
+	private String getHeightFromInput()
+	{
+		if (isMetric)
+		{
+			if (!heightInput.getText().toString().isEmpty())
+			{
+
+				return heightInput.getText().toString().replace(",", ".");
```

Comment:
Where does the "," come from?

Reply:
Ah, I see. If the user enters it with , instead of . (in Germany, e.g.) However, I think, even the German keyboard layout for numbers limits the input to `.` as a decimal separator.

Reply:
Right, this is why I added it...

> I think, even the German keyboard layout for numbers limits the input to . as a decimal separator.

I don't think so... I even specified the possible input values (the same is done for the housenumber quest too) here:
https://github.com/ENT8R/StreetComplete/blob/59b437425c51bacc9c0fd8c8b9866c86e4f474e2/app/src/main/res/layout/quest_max_height.xml#L22

Reply:
ENT8R, the line you cited is not there to allow German decimal points but to allow an input like "3,5" (multiple housenumbers).

I do not believe that Android multilingual support is supposed to be that crappy that it leaves it to the app to "understand" the number of the user input. 

You should set `android:inputType=numberDecimal` and probably remove the android:digits="0123456789.,", then everything should be fine.

Reply:
Alright. Will do that!